src/components/test-selection/TestSelection.tsx

```diff
@@ -27,53 +22,30 @@ class TestSelection extends React.Component<Props> {
     },
   ];
 
-  public render() {
-    return (
-      <div className="test-selection">
-        <Navbar
-          className="test-selection__navbar"
-          backButton={true}
-          title={i18n("nav_test")}
-          onBackButtonClick={this.onNavbarBackButtonClick}
-        />
-
-        <div className="test-selection__entries">
-          {this.testEntries.map((testEntry, index) => (
-            <TestEntry
-              key={index}
-              {...testEntry}
-              onPracticeClick={this.onTestEntryPracticeListener(testEntry)}
-              onSettingsClick={this.onTestEntrySettingsListener(testEntry)}
-            />
-          ))}
-        </div>
+  const history = useHistory();
+
+  return (
+    <div className="test-selection">
+      <Navbar
+        className="test-selection__navbar"
+        backButton={true}
+        title={i18n("nav_test")}
+        // TODO: replace "push" with "replace" in the future
```

Comment:
Really?  You want to replace the history whn using the back button?  I mean, it's you're call, but IMO replace should only be used when the user never actually visited a page or they cannot safely return to it.

Reply:
When using `push`, you're not actually going back in the history, and because of that, the back button or gesture would not work as expected.

Always using replace would not fix that issue. We need first to check if the user has previous navigations inside the app. If that's the case, we'll use `goBack`, and if not, it will use a `replace` to prevent the user from going outside the app.

That's why I left it as a TODO; it requires more than just using `replace` :sweat_smile: